EDDiscovery/TravelHistoryControl.cs

```diff
@@ -665,10 +658,11 @@ private void buttonUpdate_Click(object sender, EventArgs e)
                 distance.Store();
                 SQLiteDBClass.AddDistanceToCache(distance);
 
-                dataGridViewTravel.Rows[lastRowIndex].Cells[2].Value = textBoxDistance.Text.Trim();
+                if (dataGridViewTravel.SelectedCells.Count > 0)          // if we have selected (we should!)
+                    dataGridViewTravel.Rows[dataGridViewTravel.SelectedCells[0].OwningRow.Index].Cells[2].Value = textBoxDistance.Text.Trim();
```

Comment:
Refactoring suggestion: Lots of references to "cells[2]", "cells[3]" etc. Maybe using an enum instead of a number would aid readability and also make the code more change resistant